front-end/src/Components/Reports/reports.tsx

```diff
@@ -0,0 +1,310 @@
+import React, { useMemo } from 'react';
+import {
+  Button,
+  Column,
+  DataTable,
+  DataTableHeader,
+  DataTableRow,
+  DataTableSkeleton,
+  FormLabel,
+  Grid,
+  Pagination,
+  Row,
+  Select,
+  SelectItem,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableHeader,
+  TableRow,
+} from 'carbon-components-react';
+import {
+  getBudgets,
+  getCounties,
+  getDepartments,
+  getPrograms,
+  getProjects,
+  getReport,
+  getSites,
+  trackEmployees,
+} from './reports.resource';
+
+export const Report: React.FC = () => {
+  const [firstRowIndex, setFirstRowIndex] = React.useState(0);
+  const [currentPageSize, setCurrentPageSize] = React.useState(10);
+  const [counties, setCounties] = React.useState([]);
+  const [departments, setDepartments] = React.useState([]);
+  const [projects, setProjects] = React.useState([]);
+  const [budgets, setBudgets] = React.useState([]);
+  const [sites, setSites] = React.useState([]);
+  const [programs, setPrograms] = React.useState([]);
+  const [report, setReport] = React.useState([]);
+  const [show, setShow] = React.useState(false);
+  const [selectedValues, setSelectedValues] = React.useState({
+    department: '',
+    project: '',
+    site: '',
+    budget: '',
+    county: '',
+    contractStatus: '',
+    programArea: '',
+  });
+
+  const handleChange = (e: any) => {
+    setSelectedValues((current) => ({
+      ...current,
+      [e.target.id]: e.target.value,
+    }));
+  };
+
+  const tableHeaders: Array<DataTableHeader> = useMemo(
+    () => [
+      // { key: 'id', header: 'ID' },
+      { key: 'pfNumber', header: 'PF Number' },
+      { key: 'name', header: 'Name' },
+      { key: 'status', header: 'Status' },
+      { key: 'department', header: 'Department' },
+      { key: 'project', header: 'Project' },
+      { key: 'site', header: 'Site' },
+      { key: 'county', header: 'County' },
+      { key: 'program', header: 'Program Area' },
+    ],
+    [],
+  );
+
+  useMemo(() => {
+    getCounties().then((res) => {
+      const results = res.map((county: any) => {
+        return {
+          ...county,
+          counties: county.name,
+        };
+      });
+      setCounties(results);
+    });
+    getDepartments().then((res) => {
+      const results = res.map((department: any) => {
+        return {
+          ...department,
+          departments: department.name,
+        };
+      });
+      setDepartments(results);
+    });
+    getProjects().then((res) => {
+      const results = res.map((project: any) => {
+        return {
+          ...project,
+          projects: project.name,
+        };
+      });
+      setProjects(results);
+    });
```

Comment:
This repetition of calling individual endpoint could be combined using `Promise.all` to ensure we receive all the data we require. Please check this [documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all)